src/lib.rs

```diff
@@ -79,12 +84,45 @@ pub trait IsExecutable {
     fn is_executable(&self) -> bool;
 }
 
+/// Returns `Result<Path, io::Error>` if there is a file at the given path and the
+/// current run-level is permitted to execute it.
+///
+/// See the module documentation for details.
+#[cfg(unix)]
+pub fn is_permitted<P>(path: P) -> Result<::std::path::PathBuf, io::Error>
+where
+    P: AsRef<Path>
+{
+    path.as_ref().is_permitted()
+}
+
+/// An extension trait for `std::fs::Path` providing an `is_permitted` method.
+///
+/// See the module documentation for examples.
+#[cfg(unix)]
+pub trait IsPermitted {
```

Comment:
Both this trait declaration and `is_permitted` might as well go into the `mod unix`, You may need to add a `pub use` statement to re-export what's needed to the base of the crate if that's what you're after